### PR TITLE
Document storeCompactionState

### PR DESCRIPTION
**Pull request by ektravel**
### Document storeCompactionState context parameter

This PR:

- Rearranges the context parameters in alphabetical order.
- Add `storeCompactionState` context parameter to the table.
- Add "ingestion/tasks" back to sidebars.json (mistakenly removed on [PR 14023](https://github.com/apache/druid/pull/14023)) 

This PR has:

- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [ ] added documentation for new or modified features or behaviors.
- [ ] a release note entry in the PR description.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/dev/license.md)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [ ] been tested in a test Druid cluster.

### CODE REVIEW

**Comment by ektravel** [comment]
@suneet-s Please review the definition for `storeCompactionState` on [line 394](https://github.com/apache/druid/pull/14354/files#diff-6812a1fef2b8183b749a2cb629dd1e34bc8aeb7e520b7dee73e2a66b08caf894R394).

**Review by kfaraz — COMMENTED** [review]
Thanks for making the changes, @ektravel . I have left some comments.

File: docs/ingestion/tasks.md [hunk 1]
```diff
@@ -387,13 +387,14 @@ The settings get passed into the `context` field of the compaction tasks issued
...
+|`forceTimeChunkLock`|_Setting this to false is still experimental._<br/> Force to always use time chunk lock. If not set, each task automatically chooses a lock type to use. If set, this parameter overwrites `druid.indexer.tasklock.forceTimeChunkLock` [configuration for the overlord](../configuration/index.md#overlord-operations). See [Locking](#locking) for more details.|true|
+|`priority`|Task priority|Depends on the task type. See [Priority](#priority) for more details.|
+|`storeCompactionState`|Determines whether the task's metadata stores the state of the segments created by that task. In most cases, you should not need to set this parameter as it is set automatically on compaction tasks. |True by default for compaction tasks. For all other tasks, defaults to false. |
```
[the remark this answers is not in the document]
Comment (techdocsmith) [reply docs/ingestion/tasks.md]: We should have an official style for this however,  "whether" is not that great an introductory word. Other considerations:
> When `true`, Druid stores the compaction state...
> Enables the task to store metadata about the compaction state of created segments.

I think `true` and `false` as configuration options should probably be code terms. Except where case matters, I think they should be all lowercase.

[the remark this answers is not in the document]
Comment (ektravel) [reply docs/ingestion/tasks.md]: @suneet-s 
A compaction task launches several sub-tasks too. Can you please clarify which of the exact task types have this flag as `true`?

[the remark this answers is not in the document]
Comment (kfaraz) [reply docs/ingestion/tasks.md]: Makes sense, @suneet-s . Thanks for the clarification!

**Review by techdocsmith — APPROVED** [review]
LGTM

**Review by suneet-s — APPROVED** [review]

**Pull request merged**